samples/api/hello_triangle/hello_triangle.cpp

```diff
@@ -173,59 +169,59 @@ void HelloTriangle::init_instance(Context                         &context,
 	std::vector<VkExtensionProperties> available_instance_extensions(instance_extension_count);
 	VK_CHECK(vkEnumerateInstanceExtensionProperties(nullptr, &instance_extension_count, available_instance_extensions.data()));
 
-	std::vector<const char *> active_instance_extensions(required_instance_extensions);
+	std::vector<const char *> required_instance_extensions{VK_KHR_SURFACE_EXTENSION_NAME};
 
 #if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
-	bool has_debug_report = false;
+	// Validation layers help finding wrong api usage, we enable them when explicitly requested or in debug builds
+	// For this we use the debug utils extension if it is supported
+	bool has_debug_utils = false;
```

Comment:
Maybe instead:
```
	bool has_debug_utils = std::any_of(available_instance_extensions.begin(),
	                                   available_instance_extensions.end(),
	                                   [](auto const &ext) { return strcmp(ext.extensionName, VK_EXT_DEBUG_UTILS_EXTENSION_NAME) == 0; });
	if (has_debug_utils)
	{
		required_instance_extensions.push_back(VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
	}
	else
	{
		LOGW("{} not supported or available", VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
		LOGW("Make sure to compile the sample in debug mode and/or enable the validation layers");
	}
```

Reply:
Tbh. I'm not a fine of such single line loops. I find them hard to read. Probably a personal preference, but since this is not a Vulkan.hpp based sample, I'd prefer the loop as it currently is implemented.

Reply:
Of course, your code is equivalent to the STL algorithm, so it's fine to just keep it.
But if we want to use "Modern Vulkan", we might also want to use "Modern C++". And I think, using an STL algorithm not only is potentially more efficient and less error prone, it also tells the intent of the code sequence. You would not have to parse a loop and check if anything unexpected is happening there.
Note, with C++20, we could even use `std::ranges::any_of`, making it even more concise.

Reply:
I totally agree with that, but that's the scope of another sample to come. I'd rather not just change one place in this sample to use a different way for the loops than the others. If that's okay with you I'd keep it the way it is.

Reply:
Well, I don't quite understand your reasoning here. You change this part of the code here anyway.

Reply:
What I meant with my comment: I did not rewrite any of the loops and simply renamed a variable. Other places in the sample also use the "old style" for loops, that's why I didn't want to use a different approach in here. I'm a fan of keeping the scope of a PR as close to the problem solution as possible ;)